app/scripts/modules/cloudfoundry/src/instance/details/details.controller.ts

```diff
@@ -0,0 +1,132 @@
+import { IController, IPromise, IQService, module } from 'angular';
+
+import { StateService } from '@uirouter/angularjs';
+import { cloneDeep, flattenDeep } from 'lodash';
+
+import {
+  Application,
+  CONFIRMATION_MODAL_SERVICE,
+  ConfirmationModalService,
+  INSTANCE_WRITE_SERVICE,
+  InstanceReader,
+  InstanceWriter,
+  RecentHistoryService,
+} from '@spinnaker/core';
+
+import { ICloudFoundryInstance } from 'cloudfoundry/domain';
+
+interface InstanceFromStateParams {
+  instanceId: string;
+}
+
+interface InstanceManager {
+  account: string;
+  region: string;
+  category: string; // e.g., serverGroup, loadBalancer.
+  name: string; // Parent resource name, not instance name.
+  instances: ICloudFoundryInstance[];
+}
+
+class CloudFoundryInstanceDetailsController implements IController {
+  public state = { loading: true };
+  public instance: ICloudFoundryInstance;
+  public instanceIdNotFound: string;
+  public upToolTip = "A CloudFoundry instance is 'Up' if a load balancer is directing traffic to its server group.";
```

Comment:
You'll hit a linter error on the `"` (use a single quote instead).

Reply:
updated to suggested changes

Reply:
updated to suggested changes

Reply:
updated to suggested changes